### ISSUE TITLE
Gating issue from Cytobank ACS file

### ISSUE THREAD

**Issue by JimboMahoney** [RGLab/CytoML#73]
Hi there!

I just imported the gates from a Cytobank ACS file and I think I'm seeing something similar to #28.

Please feel free to close this issue if you believe it's the same thing (or perhaps I misunderstand that issue).

In short, all gates seem to import OK, _except_ those where the gate is positioned at the top left of the plot and overlap the Y axis.

In the below screenshot, you can see the hierarchy for the "Naive CD4" population.

![image](https://user-images.githubusercontent.com/7777844/67856528-a486fc80-fb0c-11e9-81f2-6e2c1f60b3fd.png)

The stats in CytoML are good until it hits the penultimate gate (CD4+). Both that gate, and any subsequent gates, contain a fraction of the cells they should.

The previous gate (CD3+) is fine, despite being "off-axis".

In addition, in the following screenshots:

![image](https://user-images.githubusercontent.com/7777844/67856822-50304c80-fb0d-11e9-9598-7c6759e2c7c4.png)

The CD3- population (final gate), despite overlapping both X and Y axes, is fine.

![image](https://user-images.githubusercontent.com/7777844/67856981-a4d3c780-fb0d-11e9-959a-dfff63215ded.png)

Here, the Naive CD8 (final gate) population is also fine, despite being off the chart.

This leads me to think there is something particular happening with gates that are positioned in the upper left corner (see CD4+ in first screenshot).

I hope this is clear (ish?)

I've uploaded the ACS file below.

[experiment_265121_Oct-30-2019_11-28-AM.zip](https://github.com/RGLab/CytoML/files/3788658/experiment_265121_Oct-30-2019_11-28-AM.zip)

**Comment by mikejiang** [comment]
It is due to the incorrect compensation during `cytobank_to_gatingSet`
Because in the gatingML file, `compensation-ref` is pointing to `FCS`
![image](https://user-images.githubusercontent.com/1385649/67901656-90e79000-fb24-11e9-98b5-fc2c59e37b5f.png)
Thus the parser ignores the `transforms:spectrumMatrix` node in xml, instead, it tries to extract the `spillover` matrix stored in FCS files. 
Typically we apply the same matrix across all samples and so the first FCS file is currently used as the source for that matrix. Apparently, it gets the wrong matrix since the compensation control files are also included in this experiment and the first file happens to be the control file.

There 4 options:
* compensate each file based on its own FCS-stored spillover matrix.
* use the compensation stored in xml regardless `compensation-ref` value
* exclude the control files from the parser (by looking at panel info in yaml file)
* use the compensation stored in yaml file

We don't know how cytobank works behind the scene and I am leaning toward 3rd option because it doesn't break the existing logic of gatingML-based parser. 
Let me know if it is ok with you to skip control files for the final import.

**Comment by mikejiang** [comment]
Here is the gate after the patch, let me know if it looks good to you.
```r
> ce <- open_cytobank_experiment("~/Downloads/experiment_265121_Oct-30-2019_11-28-AM.acs")
Unpacking ACS file...
> ce
cytobank Experiment:  Aria Training Sort 
gatingML File:  /tmp/RtmpNm6ewZ/file2b6230655679/experiments/265121/cytobank_gate_ml2_v2.xml 
    panel samples
1 Panel 1       5
> gs <- cytobank_to_gatingset(ce, panel_id = 1)
> library(ggcyto)
> autoplot(gs[1], "CD4+")
```
![image](https://user-images.githubusercontent.com/1385649/67907245-82a17000-fb34-11e9-955a-87e0f340f44c.png)

**Comment by JimboMahoney** [comment]
Hi Mike,

Ah, I see. So the reason the other gates I mentioned aren't affected is because they're either not affected by compensation (e.g. SSA vs CD3) or are so far out in the log scale (e.g. CD4 vs CD8, CD45RA vs. CD8) that they are not badly affected.

I can't suggest which option to go for (although the first one - use each FCS file) sound best to me as a novice, but your gate screenshot looks good.

Thanks again for super-fast responses!

**Comment by jacobpwagner** [comment]
Minor fix in 1d9751979abc3f25ee0d8b01f27bc3ee32e5635d, so make sure to grab that too @JimboMahoney
